File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_cpp_labs_and_solutions/null_object/solution/literal.cpp
// literal.cpp

class expression;
#include "expression_visitor.h"
#include "literal.h"

literal::literal(int a_value)
{
	set_value(a_value);
}

void literal::accept(expression_visitor & a_expression_visitor)
{
	a_expression_visitor.visit(*this);
}

void literal::set_value(int a_value)
{
	my_value = a_value;
}

int literal::get_value() const
{
	return my_value;
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_java_labs_and_solutions/null_object/lab/Literal.java
// Literal.java

public class Literal extends Terminal
{
	public Literal(int aValue)
	{
		setValue(aValue);
	}

	public void accept(ExpressionVisitor aExpressionVisitor)
	{
		aExpressionVisitor.visit(this);
	}

	private void setValue(int aValue)
	{
		myValue = aValue;
	}

	public int getValue()
	{
		return myValue;
	}

	private int myValue;
}
